src/Symfony/Component/Validator/Validator/RecursiveContextualValidator.php

```diff
@@ -782,8 +783,9 @@ private function validateInGroup($value, $cacheKey, MetadataInterface $metadata,
             // that constraints belong to multiple validated groups
             if (null !== $cacheKey) {
                 $constraintHash = spl_object_hash($constraint);
-
-                if ($constraint instanceof Composite) {
+                // instanceof Valid: In case of using a Valid constraint with many groups
+                // it makes a reference object get validated by each group
+                if ($constraint instanceof Composite || $constraint instanceof Valid) {
```

Comment:
@HeahDude @stof The proposed change is similar to #29499 that we discussed during SymfonyCon Lisbon. To me this looks good but I would like to get your opinion on this too.

Reply:
Indeed, looks good to me too.